tests/neg/mirror-synthesis-errors.check

```diff
@@ -0,0 +1,28 @@
+-- Error: tests/neg/mirror-synthesis-errors.scala:21:32 ----------------------------------------------------------------
+21 |val testA = summon[Mirror.Of[A]] // error: Not a sealed trait
+   |                                ^
+   |No given instance of type deriving.Mirror.Of[A] was found for parameter x of method summon in object Predef. Failed to synthesize an instance of type deriving.Mirror.Of[A]: trait A is not a sealed trait
+-- Error: tests/neg/mirror-synthesis-errors.scala:22:32 ----------------------------------------------------------------
+22 |val testC = summon[Mirror.Of[C]] // error: Does not have subclasses
+   |                                ^
+   |No given instance of type deriving.Mirror.Of[C] was found for parameter x of method summon in object Predef. Failed to synthesize an instance of type deriving.Mirror.Of[C]: trait C does not have subclasses
+-- Error: tests/neg/mirror-synthesis-errors.scala:23:32 ----------------------------------------------------------------
+23 |val testD = summon[Mirror.Of[D]] // error: child SubD takes more than one parameter list
+   |                                ^
+   |No given instance of type deriving.Mirror.Of[D] was found for parameter x of method summon in object Predef. Failed to synthesize an instance of type deriving.Mirror.Of[D]: class D's child class SubD is not a generic product because class SubD takes more than one parameter list
+-- Error: tests/neg/mirror-synthesis-errors.scala:24:38 ----------------------------------------------------------------
+24 |val testSubD = summon[Mirror.Of[SubD]] // error: takes more than one parameter list
+   |                                      ^
+   |No given instance of type deriving.Mirror.Of[SubD] was found for parameter x of method summon in object Predef. Failed to synthesize an instance of type deriving.Mirror.Of[SubD]: class SubD is not a sealed class
```

Comment:
I find this one interesting - here it first fails the isGenericProduct test, then it fails the isGenericSum test so we see the final error saying it is not a "sealed class". I think for `Synthesiser.synthesizedMirror` it could be good to prefix why it failed the `isGenericProduct`, and then also prefix `and it is not a sum because class SubD is not a sealed class`

Reply:
`SubD` _is_ a case class. In that case, I would expect a message saying “class SubD is a case class but it …” explaining what was wrong with it.

On the other hand, if a type is neither a `case class` nor a `sealed trait`, then I would expect a message saying that “SubD should be either a case class or a sealed trait”.

Reply:
we have a lot of more fine grained reasons why it can fail so I think its good to capture the errors from both branches

Reply:
I made it so it collects errors from both branches. The error for synthesis for SubD is:
```
No given instance of type deriving.Mirror.Of[SubD] was found for parameter x of method summon in object Predef. Failed to synthesize an instance of type deriving.Mirror.Of[SubD]: 
	* class SubD is not a generic sum because it is not a sealed class
	* class SubD is not a generic product because it takes more than one parameter list
   ```
I also removed the condition from `synthesizedMirror`, but `synthesizedProductMirror` makes an early exit anyway, so I think it's cleaner that way.

Reply:
I think this is very nice!